lbrynet/core/client/ClientProtocol.py

```diff
@@ -74,6 +73,7 @@ def connectionLost(self, reason):
         if self._blob_download_request is not None:
             self._blob_download_request.cancel(err)
         self._connection_manager.protocol_disconnected(self.peer, self)
+        self.factory.deferred.callback(True)
```

Comment:
self.factory.deferred is a deferred that is supposed to fire when a client connection has completely terminated (either it failed to connect, or succeeded to connect). The other places where this deferred gets fired is clientConnectionFailed() under ClientProtocolFactory. In the unit test we wait for this deferred to check if a ClientProtocol connection was sucessful or not.